syntax = "proto3";
package trade;

service TradeService {
    rpc request_trade (TradeRequest) returns (TradeResponse) {}
    rpc answer_trade (AnswerTradeRequest) returns (AnswerTradeResponse) {}
    rpc get_trades (GetTradesRequest) returns (GetTradesResponse) {}
}

message TradeRequest {
    repeated int32 my_stickers = 1;
    repeated int32 other_stickers = 2;
    string my_username = 3;
    string other_username = 4;
}

message TradeResponse {
    bool status = 1;
}

message AnswerTradeRequest {
    bool accept = 1;
}

message AnswerTradeResponse {
    bool status = 1;
}

message GetTradesRequest {
    string username = 1;
}

message GetTradesResponse {
    message Trade {
        message Sticker {
            string playername = 1;
            string country = 2;
            int32 rarity = 3;
            int32 id = 4;
        }

        enum Status {
            PENDENT = 0;
            ACCEPTED = 1;
            RECUSED = 2;
        }
    
        repeated Sticker to_send = 1;
        repeated Sticker to_receive = 2;
        string username = 3;
        Status status = 4;
        int32 trade_id = 5;
    }

    repeated Trade trades = 1;
}